src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -37,11 +39,29 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
         private List<MediaTypeHeaderValue> GetContentTypes(string firstArg, string[] args)
```

Comment:
Assuming you're referring to 
```if (contentType.MatchesAllSubTypes || contentType.MatchesAllTypes || contentType.Type == "*")```

**Ideal** - There's something on the `MediaTypeHeaderValue` that does what we need
**Less Ideal** - Make a static method somewhere that has logic

If you're referring to the general process of combining args and calling the parse method, that's just boilerplate, leave it as-is imo.